tests/src/test/scala/cats/tests/ContTSuite.scala

```diff
@@ -86,4 +86,27 @@ class ContTSuite extends CatsSuite {
     }
   }
 
+  test("ContT.resetT and shiftT delimit continuations") {
+    forAll { (cb: Unit => Eval[Unit]) =>
+      var counter = 0
+
+      val contT: ContT[Eval, Unit, Unit] = ContT.resetT(
+        ContT.shiftT { (k: Unit => Eval[Unit]) =>
+          ContT.defer[Eval, Unit, Unit] {
+            counter = counter + 1
+          } >>
+            ContT.liftF(k(())) >>
```

Comment:
I have no intuition for why this is testing anything when it is true for all cbs.

Reply:
Yes, good point 😂 I'll try to come up with a better test. Really we want to assert on the evaluation order, because that is different from other monads but it's not so obvious to do that. I guess we could have multiple `var`s and assign them a monotonic timestamp as we evaluate?

Reply:
Actually just realized `AtomicInteger#incrementAndGet()` would be much better